crates/indexer/src/helper.rs

```diff
@@ -117,18 +83,47 @@ pub enum GwLog {
     },
 }
 
-fn parse_sudt_log_data(data: &[u8]) -> ([u8; 20], [u8; 20], u128) {
-    assert_eq!(data[0], 20);
-    let mut from_address = [0u8; 20];
-    from_address.copy_from_slice(&data[1..21]);
+fn parse_sudt_log_data(data: &[u8]) -> anyhow::Result<(RegistryAddress, RegistryAddress, u128)> {
+    let mut start = 0;
+    let mut end = 0;
```

Comment:
assert!(data.len() >= 8);

Reply:
Actually, data length has been checked before enter this function